Sources/BraintreeLocalPayment/Models/LocalPaymentPOSTBody.swift

```diff
@@ -0,0 +1,119 @@
+        self.experienceProfile = ExperienceProfile(
+            noShipping: !localPaymentRequest.isShippingAddressRequired,
+            brandName: localPaymentRequest.displayName
+        )
+        self.intent = "sale"
+        self.returnURL = BTCoreConstants.callbackURLScheme + "://x-callback-url/braintree/local-payment/success"
+        self.cancelURL = BTCoreConstants.callbackURLScheme + "://x-callback-url/braintree/local-payment/cancel"
+    }
+    
+    enum CodingKeys: String, CodingKey {
+        case paymentType = "funding_source"
+        case amount
+        case currencyCode = "currency_iso_code"
+        case paymentTypeCountryCode = "payment_type_country_code"
+        case merchantAccountID = "merchant_account_id"
+        case email = "payer_email"
+        case givenName = "first_name"
+        case surname = "last_name"
+        case phone
+        case bic
+        case intent
+        case returnURL = "return_url"
+        case cancelURL = "cancel_url"
+        case experienceProfile = "experience_profile"
+        
+        // Address keys
+        case streetAddress = "line1"
+        case extendedAddress = "line2"
+        case locality = "city"
+        case countryCodeAlpha2 = "country_code"
+        case postalCode = "postal_code"
+        case region = "state"
+    }
+    
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(paymentType, forKey: .paymentType)
+        try container.encode(amount, forKey: .amount)
+        try container.encode(currencyCode, forKey: .currencyCode)
+        try container.encodeIfPresent(paymentTypeCountryCode, forKey: .paymentTypeCountryCode)
+        try container.encodeIfPresent(merchantAccountID, forKey: .merchantAccountID)
+        try container.encodeIfPresent(email, forKey: .email)
+        try container.encodeIfPresent(givenName, forKey: .givenName)
+        try container.encodeIfPresent(surname, forKey: .surname)
+        try container.encodeIfPresent(phone, forKey: .phone)
+        try container.encodeIfPresent(bic, forKey: .bic)
+        try container.encodeIfPresent(intent, forKey: .intent)
+        try container.encodeIfPresent(returnURL, forKey: .returnURL)
+        try container.encodeIfPresent(cancelURL, forKey: .cancelURL)
+        try container.encodeIfPresent(experienceProfile, forKey: .experienceProfile)
+          
+        if let address {
+            try container.encodeIfPresent(address.streetAddress, forKey: .streetAddress)
+            try container.encodeIfPresent(address.extendedAddress, forKey: .extendedAddress)
+            try container.encodeIfPresent(address.locality, forKey: .locality)
+            try container.encodeIfPresent(address.countryCodeAlpha2, forKey: .countryCodeAlpha2)
+            try container.encodeIfPresent(address.postalCode, forKey: .postalCode)
+            try container.encodeIfPresent(address.region, forKey: .region)
+        }
```

Comment:
Take it or leave it - instead of needing a custom Encoder implementation, we could have the `init` take in a `BTPostalAddress` param, but then the private properties on this struct are the actual encodable types.

Roughly something like:
```
private let amount: String
private var streetAddress: String?
private var extendedAddress: String?

init(amount: String, ... the other stuff, address: BTPostalAddress?) {
    self.amount = amount 
    self.streetAddress = address.streetAddress
    self.extendedAddress = address.extendedAddress
}
```

Then we wouldn't need to override `encode()`.

Which is more readable? I go back and forth sometimes with the custom encoder being more or less explicit.

Reply:
Updated: [3e00b0e](https://github.com/braintree/braintree_ios/pull/1468/commits/3e00b0e96d9cd3d3ca2e5601a1bc8cf79acb3d31)